docs/best-practices.rst

```diff
@@ -0,0 +1,271 @@
+ )
+
+Deployment in Production
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Once you have completed all the mentioned checks, it is time to deploy your DAG in production.
+To do this, first, you need to make sure that the Airflow is itself production-ready. 
+Let's see what precautions you need to take.
+
+
+Backend
+--------
+
+Airflow comes with an ``SQLite`` backend by default. It allows the user to run Airflow without any external database.
+However, such a setup is meant to be for testing purposes only. Running the default setup can lead to data loss in multiple scenarios. 
+If you want to run Airflow in production, make sure you :doc:`configure the backend <howto/initialize-database>` to be an external database such as ``MySQL`` or ``Postgres``. 
+
+You can change the backend using the following config-
+
+.. code::
+
+ [core]
+ sql_alchemy_conn = my_conn_string
+
+Once you have changed the backend, airflow needs to create all the tables required for operation.
+Create an empty DB and give airflow's user the permission to ``CREATE/ALTER`` it.
+Once that is done, you can run -
+
+.. code::
+
+ airflow upgradedb
+
+``upgradedb`` keeps track of migrations already applies, so it's safe to run as often as you need.
+
+.. note::
+ 
+ Do not use ``airflow initdb`` as it can create a lot of default connection, charts, etc. which are not required in production DB.
+
+
+Multi-Node Cluster
+-------------------
+
+Airflow uses :class:`airflow.executors.sequential_executor.SequentialExecutor` by default. It works fine in most cases. However, by its nature, the user is limited to executing at most
+one task at a time. It's also not suitable to work in a multi-node cluster. You should use :doc:`../executor/celery` or :doc:`../executor/kubernetes` in such cases.
+
+
+Once you have configured the executor, it is necessary to make sure that every node in the cluster contains the same configuration and dags.
+Airflow only sends simple instructions such as execute task X on node Y but does not send any dag files or configuration. You can use a simple CRON or
+any other mechanism to sync DAGs and configs across your nodes, e.g., checkout DAGs from git repo every 5 minutes on all nodes.
+
+
+Logging
+--------
+
+If you are using disposable nodes in your cluster, configure the log storage to be a distributed file system such as ``S3`` or ``GCS``.
+A DFS makes these logs are available even after the node goes down or gets replaced. See :doc:`howto/write-logs` for configurations.
+
+.. note::
+
+    The logs only appear in your DFS after the task has finished. You can view the logs while the task is running in UI itself.
+
+
+Configuration
+--------------
+
+Airflow comes bundles with a default airflow.cfg configuration file.
+You should environment variables for configurations that change across deployments
+e.g. metadata DB, password. You can do it using the format ``$AIRFLOW__{SECTION}__{KEY}``
+
+.. code::
+
+ AIRFLOW__CORE__SQL_ALCHEMY_CONN=my_conn_id
```

Comment:
There's also the option of using `sql_alchemy_conn_cmd` (only for the connection, not everything has a `_cmd` "flavour") so that you don't have to put secrets in the config file.